idaes/core/util/model_diagnostics.py

```diff
@@ -4122,3 +4264,269 @@ def _collect_model_statistics(model):
+                f"({str(node)})."
+            )
+
+        return [numerator / denominator], mismatch, cancelling, const
+
+    def _check_power(self, node, child_data):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+
+        base = self._get_value_for_sum_subexpression(child_data[0])
+        exponent = self._get_value_for_sum_subexpression(child_data[1])
+
+        return [base**exponent], mismatch, cancelling, const
+
+    def _check_negation(self, node, child_data):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+        val = -self._get_value_for_sum_subexpression(child_data[0])
+
+        return [val], mismatch, cancelling, const
+
+    def _check_abs(self, node, child_data):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+        val = abs(self._get_value_for_sum_subexpression(child_data[0]))
+
+        return [val], mismatch, cancelling, const
+
+    def _check_unary_function(self, node, child_data):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+
+        func_name = node.getname()
+        func = getattr(math, func_name)
+        func_val = self._get_value_for_sum_subexpression(child_data[0])
+
+        try:
+            val = func(func_val)
+        except ValueError:
+            raise ValueError(
+                f"Error in ConstraintTermAnalysisVisitor: error evaluating {str(node)}."
+            )
+
+        return [val], mismatch, cancelling, const
+
+    def _check_other_expression(self, node, child_data):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+
+        # First, need to get value of input terms, which may be sub-expressions
+        input_mag = [self._get_value_for_sum_subexpression(i) for i in child_data]
+
+        # Next, create a copy of the external function with expected magnitudes as inputs
+        newfunc = node.create_node_with_local_data(input_mag)
+
+        # Evaluate new function and return the absolute value
+        return [value(newfunc)], mismatch, cancelling, const
+
+    def _perform_checks(self, node, child_data):
+        # Perform checks for problematic expressions
+        # First, need to check to see if any child data is a list
+        # This indicates a sum expression
+        mismatch = []
+        cancelling = []
+        const = True
+
+        for d in child_data:
+            # Collect all warnings about mismatched terms from child nodes
+            mismatch += d[1]
+            cancelling += d[2]
+
+            # We will check for cancelling terms here, rather than the sum itself, to handle special cases
+            # We want to look for cases where a sum term results in a value much smaller
+            # than the terms of the sum
+            sums = self._sum_combinations(d[0])
+            if any(i <= self._sum_tol * max(d[0]) for i in sums):
+                cancelling.append(str(node))
```

Comment:
I think I understand what you are doing here, but wouldn't this complain loudly about the objective for every parameter estimation problem (MSE) if the problem actually solved?

Reply:
One of my main concerns with this tool is its inability to distinguish between true cases of catastrophic cancellation and benign situations, where, for example, we have a heat of adsorption term in a column which will be close to except near the breakthrough point.

Reply:
Is there an easy way to test for that however? Note that this tool is only used for Cautions in the main toolbox, with the implication that these might be issues (but not guaranteeing that).

I.e., this is intended to be a simple check to find potential issues, but the user will have to look into them all further to decide if they are critical or not.

Reply:
@jsiirola This walker was written specifically with Constraints in mind, so I had not considered that. The check for mismatched terms would make sense for an Objective however, so this could be extended to handle them as well. However, how would the expression walker know if it was dealing with an Objective or a Constraint (the input argument is the expression, not the component)?